src/libraries/Microsoft.PowerFx.Core/Public/Config/ReadOnlySymbolTable.cs

```diff
@@ -402,6 +402,24 @@ IEnumerable<TexlFunction> INameResolver.LookupFunctionsInNamespace(DPath nameSpa
             return _functions.WithNamespace(nameSpace);
         }
 
+        internal virtual void EnumerateNames(List<SymbolEntry> names, EnumerateNamesOptions opts)
+        {
+            if (this is IGlobalSymbolNameResolver globals)
+            {
+                foreach (var variable in globals.GlobalSymbols)
+                {
+                    if (variable.Value.TryToSymbolEntry(out var x))
+                    {
+                        names.Add(x);
+                    }
+                }
+            }
+        }
+
+        internal class EnumerateNamesOptions

```

Comment:
>EnumerateNamesOptions [](http://example.com/codeflow?start=23&length=21)

How is this used?

Reply:
yes, I wanted future options to be easy, since this is a policy-driven enumeration. 
This is internal, so less penalty to have it.